### PR TITLE
Improve survey presets drop down

### PR DESCRIPTION
**Pull request by dakejahl**
This PR implements the suggested improvements to the Survey Presets UI as discussed in 
https://github.com/mavlink/qgroundcontrol/issues/8347

survey_presets

And the wizard as well
survey_presets_wizard

### CODE REVIEW

File: src/MissionManager/ComplexMissionItem.cc [hunk 1]
```diff
@@ -37,7 +37,11 @@ QStringList ComplexMissionItem::presetNames(void)
 
     settings.beginGroup(presetsSettingsGroup());
     settings.beginGroup(_presetSettingsKey);
-    return settings.childKeys();
+
+    names << "Select a preset";
```
Comment (Antiheavy) [inline src/MissionManager/ComplexMissionItem.cc]: I wonder if it would add some clarity if this text was _italic_?

Comment (dakejahl) [reply src/MissionManager/ComplexMissionItem.cc]: Yeah I wasn't quite sure how to do it, I found one way but it was a little bit hacky feeling.

**Comment by Antiheavy** [comment]
This is working awesome.  It is more intuitive in my opinion.

@dakejahl  There is still a problem where, staying inside the Survey mission item the whole time, you can select a Preset, switch to the Grid tab, change some settings, and then go back to the Presets tab and it still shows the Preset you selected (which is now no longer true because you might have changed some things).  The drop down should revert back to "Select a preset" any time you switch into that tab (because you don't know what the user might have changed while away).

The button sizing could use some tweaking.  There is some awkward spacing as it stand now.  I suggest side-by-side buttons for New Preset and Delete Preset:
image

**Comment by DonLakeFlyer** [comment]
> New Preset

That is way bad as well. The current button text helps explain to the user what a Preset is. Whereas "New" has completely different connotations as to what it may be doing. 

I don't see any of this as being better than what is there now. My reasons in the discussion issue. This lowers the usability of the product to remove one click.

**Pull request closed without merging**